src/stream.c

```diff
@@ -1056,15 +1058,20 @@ void _uvc_populate_frame(uvc_stream_handle_t *strmh) {
     frame->step = 0;
     break;
   }
-  
+
+  frame->sequence = strmh->hold_seq;
+  /** @todo set the frame time */
+  // frame->capture_time
+
   /* copy the image data from the hold buffer to the frame (unnecessary extra buf?) */
   if (frame->data_bytes < strmh->hold_bytes) {
     frame->data = realloc(frame->data, strmh->hold_bytes);
-    frame->data_bytes = strmh->hold_bytes;
   }
+  frame->data_bytes = strmh->hold_bytes;
```

Comment:
I don't like that this'll cause additional reallocations as the image size fluctuates, but I do see how you wouldn't want leftover junk bytes to get mixed in when you're dealing with compressed images. Maybe uvc_frames should have both length (used bytes) and capacity (allocated bytes)...

Anyway, I'll go with the approach you've taken here. The whole buffering situation needs an overhaul. I'll plan to take the split-length/capacity approach at some point, whenever I get around to fixing the way-oversized buffers.